### PR TITLE
[flutter_markdown] fixes null check operator used on null value if onSelectionChanged is…

### PR DESCRIPTION
**Pull request by jimmyff**
- Fixes flutter/flutter/issues/149817
- Fixes flutter/flutter/issues/146282

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing. (There is an issue with a pixel test however this minor change is unrealted)

### CODE REVIEW

**Comment by jimmyff** [comment]
As advised by navaronbracke at Discord I'll add a widget test for this and resubmit

File: packages/flutter_markdown/CHANGELOG.md
Comment (domesticmouse) [inline packages/flutter_markdown/CHANGELOG.md:159]: Can you revert the format change? It makes this PR noisy?

**Review by domesticmouse — APPROVED** [review]
LGTM once the changelog format change is reverted

File: packages/flutter_markdown/CHANGELOG.md
Comment (jimmyff) [reply packages/flutter_markdown/CHANGELOG.md:159]: @domesticmouse I had noticed the reformat had accidentally slipped through and I was going to revert it when I added the test however I realised it had only reformatted the second half of the file (0.6.10 and earlier) as the first half of the changelog was already formatted according to best practices. So it made the changelogs second half consistent with the first which I thought maybe wasn't a bad thing?

Happy to revert it though if you think keeping the pr tidy is better?  Let me know and I'll update it the same day. Thanks

Comment (domesticmouse) [reply packages/flutter_markdown/CHANGELOG.md:159]: I'll defer to @stuartmorgan on this one

**Review by domesticmouse — APPROVED** [review]
PTAL @stuartmorgan

**Comment by domesticmouse** [comment]
Also, please fix conflicts. Thanks!

File: packages/flutter_markdown/CHANGELOG.md
Comment (domesticmouse) [inline packages/flutter_markdown/CHANGELOG.md:7]: This should be the 0.7.3 change, yeah?

File: packages/flutter_markdown/CHANGELOG.md
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: "Issue" is very vague; please make the comment specific about what is fixed.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by jimmyff** [comment]
Thanks for the feedback @stuartmorgan and @domesticmouse -I appreciate your time. 

This _should_ be good to go now but let me know if there's anything else.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**
